vm/src/vm/runners/cairo_pie.rs

```diff
@@ -64,20 +81,225 @@ pub enum BuiltinAdditionalData {
     Output(OutputBuiltinAdditionalData),
     // Signatures are composed of (r, s) tuples
     #[serde(serialize_with = "serde_impl::serialize_signature_additional_data")]
-    Signature(HashMap<Relocatable, (Felt252, Felt252)>),
+    Signature(SignatureBuiltinAdditionalData),
     None,
 }
 
-#[derive(Serialize, Clone, Debug, PartialEq, Eq)]
+#[derive(Serialize, Deserialize, Clone, Debug, Default, PartialEq, Eq)]
+pub struct AdditionalData {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub output_builtin: Option<OutputBuiltinAdditionalData>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub pedersen_builtin: Option<Vec<Relocatable>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub ecdsa_builtin: Option<SignatureBuiltinAdditionalData>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub range_check_builtin: Option<()>,
+}
```

Comment:
Is this struct necessary?

Reply:
It's necessary because serde struggles with untagged unions and hashmaps where the key is not a string. This new struct enforces the additional data type for each builtin, guaranteeing type safety and (probably) makes deserialization faster by not having to guess.

Reply:
An we have a more specific name since the struct is public? `AdditionalData` could literally be anything.